source/dlangbot/app.d

```diff
@@ -185,10 +185,26 @@ void handlePR(string action, PullRequest* _pr)
 
     pr.updateGithubComment(comment, action, refs, descs, msgs);
 
-    if (refs.any!(r => r.fixed) && comment.body_.length == 0)
+    if (refs.any!(r => r.fixed))
     {
-        logDebug("[github/handlePR](%s): adding bug fix label", _pr.pid);
-        pr.addLabels(["Bug fix"]);
+        import std.algorithm : canFind, filter, map, sort, uniq;
+        import std.array : array;
+        // references are already sorted by id
+        auto bugzillaIds = refs.map!(r => r.id).uniq;
+        auto bugzillSeverities = descs
+            .filter!(d => bugzillaIds.canFind(d.id))
+            .map!(i => i.severity)
+            .array
+            .sort()
+            .uniq;
+        logDebug("[github/handlePR](%s): trying to add bug fix label", _pr.pid);
+        string[] labels;
+        if (bugzillSeverities.canFind("enhancement"))
+            labels ~= "Enhancement";
+        else
+            labels ~= "Bug fix";
```

Comment:
Hmm, it seams that `.array.sort.uniq.array` may be needed after all.
That's because I think we need to add "Bug fix" label if there are more than one bugs attached to a PR and at least one of those bugs is not an enhancement request. Probably quite rare, and that's why I think it's worth labeling them.

```D
if (bugzillSeverities.canFind("enhancement"))
    labels ~= "Enhancement";
if (bugzillSeverities.length > 2)
    labels ~= "Bug fix";
```